plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -71,16 +71,34 @@ function perflab_query_plugin_info( string $plugin_slug ) {
 		return new WP_Error( 'no_plugins', __( 'No plugins found in the API response.', 'performance-lab' ) );
 	}
 
-	$plugins            = array();
-	$standalone_plugins = array_merge(
-		array_flip( perflab_get_standalone_plugins() ),
-		array( 'optimization-detective' => array() ) // TODO: Programmatically discover the plugin dependencies and add them here. See <https://github.com/WordPress/performance/issues/1616>.
-	);
-	foreach ( $response->plugins as $plugin_data ) {
-		if ( ! isset( $standalone_plugins[ $plugin_data['slug'] ] ) ) {
+	$plugins      = array();
+	$plugin_queue = perflab_get_standalone_plugins();
+
+	// Index the plugins from the API response by their slug for efficient lookup.
+	$all_performance_plugins = array_column( $response->plugins, null, 'slug' );
+
+	// Start processing the plugins using a queue-based approach.
+	while ( count( $plugin_queue ) > 0 ) { // phpcs:ignore Squiz.PHP.DisallowSizeFunctionsInLoops.Found
+		$current_plugin_slug = array_shift( $plugin_queue );
+
+		if ( isset( $plugins[ $current_plugin_slug ] ) ) {
 			continue;
 		}
-		$plugins[ $plugin_data['slug'] ] = wp_array_slice_assoc( $plugin_data, $fields );
+
+		if ( ! isset( $all_performance_plugins[ $current_plugin_slug ] ) ) {
+			return new WP_Error(
+				'plugin_not_found',
+				__( 'Plugin not found in Performance Lab\'s features / plugins.', 'performance-lab' )
```

Comment:
WordPress Core discourages unnecessary escaping in strings, so this would be better as:
```suggestion
				__( "Plugin not found in Performance Lab's features / plugins.", 'performance-lab' )
```

Also I think we shouldn't use this "features / plugins" syntax, it reads weird. Maybe better as follows? It also avoids the apostrophe.
```suggestion
				__( 'Plugin not found in WordPress.org API response.', 'performance-lab' )
```